nixos/modules/security/auditd.nix

```diff
@@ -4,15 +4,82 @@
   pkgs,
   ...
 }:
+let
+  cfg = config.security.auditd;
 
+  prepareConfigValue =
+    v:
+    if lib.isBool v then
+      (if v then "yes" else "no")
+    else if lib.isList v then
+      lib.concatStringsSep " " (map prepareConfigValue v)
+    else
+      builtins.toString v;
+  prepareConfigText =
+    conf:
+    lib.concatLines (
+      lib.mapAttrsToList (k: v: if v == null then "#${k} =" else "${k} = ${prepareConfigValue v}") conf
+    );
+in
 {
-  options.security.auditd.enable = lib.mkEnableOption "the Linux Audit daemon";
+  options.security.auditd = {
+    enable = lib.mkEnableOption "the Linux Audit daemon";
 
-  config = lib.mkIf config.security.auditd.enable {
+    config = lib.mkOption {
+      type = lib.types.submodule {
+        freeformType =
+          with lib.types;
+          attrsOf (
+            nullOr (oneOf [
+              bool
+              nonEmptyStr
+              path
+              int
+            ])
+          );
+        options = {
+          # space_left needs to be larger than admin_space_left, yet they default to be the same if left open.
+          space_left = lib.mkOption {
+            type = lib.types.either lib.types.int lib.types.nonEmptyStr;
+            default = 75;
+            description = ''
+              This is a numeric value in megabytes that tells the audit daemon when to perform a configurable action because the system is running
+              low on disk space. This should be considered the last chance to do something before running out of disk space. The numeric value for
+              this parameter should be lower than the number for space_left. You may also append a percent sign (e.g. 1%) to the number to have
+              the audit daemon calculate the number based on the disk partition size.
+            '';
```

Comment:
> should be lower than the number for space_left

The description is self-referential here.

Reply:
Oh crap, good catch! I flipped the descriptions for space_left/admin_space_left